src/ol_infrastructure/applications/starburst/__main__.py

```diff
@@ -0,0 +1,645 @@
+                for privilege_type, roles in normalized_global.items():
+                    if privilege_type not in grants_by_domain[domain_name]:
+                        grants_by_domain[domain_name][privilege_type] = []
+                    grants_by_domain[domain_name][privilege_type].extend(roles)
+                    # Remove duplicates
+                    grants_by_domain[domain_name][privilege_type] = list(
+                        set(grants_by_domain[domain_name][privilege_type])
+                    )
+
+        return grants_by_domain
+
+    def _normalize_grants(self, grants: dict[str, Any]) -> dict[str, list[str]]:
+        """Normalize grants to standard privilege names."""
+        normalized = {}
+
+        # Map dbt grant names to Starburst privilege names
+        privilege_mapping = {
+            "select": "Select",
+            "insert": "Insert",
+            "update": "Update",
+            "delete": "Delete",
+            "all privileges": "All",
+        }
+
+        for dbt_privilege, roles in grants.items():
+            starburst_privilege = privilege_mapping.get(
+                dbt_privilege.lower(), dbt_privilege
+            )
+            if isinstance(roles, list):
+                normalized[starburst_privilege] = roles
+            elif isinstance(roles, str):
+                normalized[starburst_privilege] = [roles]
+
+        return normalized
+
+    def get_data_domains(self) -> dict[str, list[str]]:
+        """Get data domains with their corresponding schema names from dbt project."""
+        domain_to_schemas: dict[str, list[str]] = {}
+        models_config = self.project_config.get("models", {})
+        project_name = self.project_config.get("name", "open_learning")
+        project_models = models_config.get(project_name, {})
+
+        # Domain mapping: dbt model key -> domain_name
+        domain_mappings = {
+            "staging": "staging",
+            "intermediate": "intermediate",
+            "marts": "mart",  # ← Key difference: plural -> singular
+            "dimensional": "dimensional",
+            "external": "external",
+            "reporting": "reporting",
+            "migration": "migration",
+        }
+
+        for dbt_key, domain_name in domain_mappings.items():
+            model_config = project_models.get(dbt_key, {})
+            if isinstance(model_config, dict) and "+schema" in model_config:
+                base_schema = model_config["+schema"]
+                domain_to_schemas[domain_name] = [
+                    f"ol_warehouse_production_{base_schema}",
+                    f"ol_warehouse_qa_{base_schema}",
+                ]
+
+        # Add 'raw' domain which is not defined in dbt models section
+        if "raw" not in domain_to_schemas:
+            domain_to_schemas["raw"] = [
+                "ol_warehouse_production_raw",
+                "ol_warehouse_qa_raw",
+            ]
+
+        return domain_to_schemas
```

Comment:
This seems like it would make sense in a `lib.data.<foobar>` module. I'm not sure exactly where else we would use it, but I can imagine having other contexts where it would be helpful to introspect details about our dbt project(s).